packages/lib/modules/pool/actions/create/steps/details/useValidatePoolHooksContract.ts

```diff
@@ -0,0 +1,124 @@
+import { useQuery } from '@tanstack/react-query'
+import { Address, isAddress, zeroAddress } from 'viem'
+import { usePublicClient } from 'wagmi'
+import { usePoolCreationForm } from '../../PoolCreationFormProvider'
+
+export const useValidatePoolHooksContract = (address: Address | undefined) => {
+  const {
+    poolConfigForm: { setValue },
+  } = usePoolCreationForm()
+  const publicClient = usePublicClient()
```

Comment:
use `useReadContract`?

Reply:
lol ya the combo of tanstack + public client rather silly here thank you 🙏